frontend/src/citizen-frontend/calendar/reservation-modal/RepetitionTimeInputGrid.tsx

```diff
@@ -0,0 +1,69 @@
+// SPDX-FileCopyrightText: 2017-2022 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React from 'react'
+import styled from 'styled-components'
+
+import FiniteDateRange from 'lib-common/finite-date-range'
+import { DailyReservationData } from 'lib-common/generated/api-types/reservations'
+import {
+  Repetition,
+  ReservationFormDataForValidation,
+  ValidationResult
+} from 'lib-common/reservations'
+import { assertNever } from 'lib-common/utils/assertNever'
+import { defaultMargins } from 'lib-components/white-space'
+
+import DailyRepetitionTimeInputGrid from './DailyRepetitionTimeInputGrid'
+import IrregularRepetitionTimeInputGrid from './IrregularRepetitionTimeInputGrid'
+import WeeklyRepetitionTimeInputGrid from './WeeklyRepetitionTimeInputGrid'
+
+export interface RepetitionTimeInputGridProps {
+  formData: ReservationFormDataForValidation
+  childrenInShiftCare: boolean
+  includedDays: number[]
+  updateForm: (updated: Partial<ReservationFormDataForValidation>) => void
+  showAllErrors: boolean
+  existingReservations: DailyReservationData[]
+  validationResult: ValidationResult
+  selectedRange: FiniteDateRange
+}
+
+export default React.memo(function RepetitionTimeInputGrid({
+  repetition,
+  ...props
+}: RepetitionTimeInputGridProps & { repetition: Repetition }) {
+  switch (repetition) {
+    case 'DAILY':
+      return (
+        <TimeInputGrid>
+          <DailyRepetitionTimeInputGrid {...props} />
+        </TimeInputGrid>
+      )
+    case 'WEEKLY':
+      return (
+        <TimeInputGrid>
+          <WeeklyRepetitionTimeInputGrid {...props} />
+        </TimeInputGrid>
+      )
+    case 'IRREGULAR':
+      return (
+        <TimeInputGrid>
+          <IrregularRepetitionTimeInputGrid {...props} />
+        </TimeInputGrid>
+      )
+    default:
+      assertNever(repetition)
+      return null
```

Comment:
I don't see the value of this default block. Typescript can already check without it that all cases are handled.